src/main/java/com/google/sps/servlets/EditEventServlet.java

```diff
@@ -0,0 +1,304 @@
+    Date startDate = parseInputDate(requestStartDate, json);
+    Date endDate = parseInputDate(requestEndDate, json);
+
+    boolean goodDateTimes = verifyDateTimes(startDateTime, endDateTime, json);
+    if (goodDateTimes) {
+      editEntity(
+        request, 
+        endDateTime, 
+        startDateTime, 
+        startDate, 
+        endDate, 
+        json, 
+        requestStartTime, 
+        requestEndTime, 
+        requestStartDate,
+        requestEndDate,
+        timeZone);  
+    }
+
+    response.setContentType("application/json;");
+    response.getWriter().println(json);
+  }
+
+/**
+  HTML Date inputs return a String in the form: yyyy/mm/dd.
+  In order to use that for comparison purposes, the String needs to be parsed 
+  into a Date, which requires taking substrings of the year, month, and day of
+  the inputted date String. 
+
+  input.substring(0, 4) should return the yyyy element of the input.
+  input.substrng(5, 7) should return the mm element of the input.
+  input.substring(8) should return the dd element of the input.
+
+  Using these pieces of the input the date can be parsed and formatted as desired.
+*/
+  private Date parseInputDateTime(String inputDate, String time, String timeZone,
+      JSONObject json) {
+    String[] splitString = inputDate.split("-");
+    String year = splitString[0];
+    if (Integer.parseInt(year) > 5000) {
+      json.put("weird-year", year);
+    }
+    else {
+      json.put("weird-year", "no");
+    }
+    String month = splitString[1];
+    String day = splitString[2];
+
+    return createDateTime(year, month, day, time, timeZone);
+  }
+
+  private Date parseInputDate(String inputDate, JSONObject json) {
+    String[] splitString = inputDate.split("-");
+    String year = splitString[0];
+    String month = splitString[1];
+    String day = splitString[2];
+
+    return createDate(year, month, day);
+  }
+
+/** Turns inputted strings into a date.*/
+  private Date createDate(String year, String month, String day) {
+    String dateString = month + "-" + day + "-" + year;
+    SimpleDateFormat formatter = new SimpleDateFormat("MM-dd-yyyy");
+    Date dateTime = new Date();
+    try {
+      dateTime = formatter.parse(dateString);
+    } catch (ParseException e) {e.printStackTrace();}
+    return dateTime;
+  }
+
+/** Turns inputted strings into date time. */
+  private Date createDateTime(String year, String month, String day, 
+            String time, String timeZone) {
+    String dateString = month + "-" + day + "-" + year + " " + time + " " + timeZone;
+    SimpleDateFormat formatter = new SimpleDateFormat("MM-dd-yyyy HH:mm z");
+    formatter.setTimeZone(TimeZone.getTimeZone(timeZone));
+    Date dateTime = new Date();
+    try {
+      dateTime = formatter.parse(dateString);
+    } catch (ParseException e) {e.printStackTrace();}
+    return dateTime;
+  }
+
+/** Creates an event entity in datastore. */
+  private void editEntity(
```

Comment:
FYI (no changes requested)

It looks like your implementation updates the complete representation of the event in your datastore. This is semantically in line with the HTTP PUT verb.

Another common pattern in web apps is to implement a PATCH action, which accepts a partial representation of the resource and only updates the fields specified in the request.